### ISSUE TITLE
mips: cone search always returns empty array

### ISSUE THREAD

**Issue by lpsinger** [astropy/astropy-healpix#86]
The tests fail on mips because `cone_search_skycoord` and `cone_search_lonlat` always return an empty array. See the [Debian build log](https://buildd.debian.org/status/fetch.php?pkg=astropy-healpix&arch=mips&ver=0.2-3&stamp=1526621225&raw=0). Here is an excerpt:

```
=================================== FAILURES ===================================
_______________________ test_healpix_cone_search[nested] _______________________

order = 'nested'

    @pytest.mark.parametrize('order', ['nested', 'ring'])
    def test_healpix_cone_search(order):
        indices = healpix_cone_search(10 * u.deg, 20 * u.deg, 1 * u.deg,
                                      nside=256, order=order)
    
>       assert len(indices) == 80
E       assert 0 == 80
E        +  where 0 = len(array([], dtype=int64))

astropy_healpix/tests/test_core.py:292: AssertionError
```

The Debian build ran on an actual mips board and I confirmed it under emulation with qemu.

**Comment by astrofrog** [comment]
@lpsinger - is there any way you know of that we could make a docker image that uses qemu to provide a mips emulated environment? It seems that e.g. here and in astropy core, this would be really useful for issues like this. I imagine setting up such a docker image might be difficult though?

**Comment by astrofrog** [comment]
(if we could set one up we could run it on CircleCI)

**Comment by lpsinger** [comment]
> @lpsinger - is there any way you know of that we could make a docker image that uses qemu to provide a mips emulated environment? It seems that e.g. here and in astropy core, this would be really useful for issues like this. I imagine setting up such a docker image might be difficult though?

No, it wouldn't really be difficult. I can certainly send you the image that I used to check this.

**Comment by olebole** [comment]
The problem is a type mismatch for  the function `healpix_rangesearch_radec_simple()`

* It is defined as `int64_t` in [`healpix_utils.c` ](https://github.com/astropy/astropy-healpix/blob/528292bc2bff01563d6c696880020345cde88d40/cextern/astrometry.net/healpix-utils.c#L142-L143),
* but declared as `int` in 
  - [`healpix-utils.h`](https://github.com/astropy/astropy-healpix/blob/528292bc2bff01563d6c696880020345cde88d40/cextern/astrometry.net/healpix-utils.h#L18) and
  - [`core_cython.pxd`](https://github.com/astropy/astropy-healpix/blob/528292bc2bff01563d6c696880020345cde88d40/astropy_healpix/core_cython.pxd#L41).
* When it is used in [`core_cython.pyx`](https://github.com/astropy/astropy-healpix/blob/528292bc2bff01563d6c696880020345cde88d40/astropy_healpix/core_cython.pyx#L485), the function is assumed to return an `int` (32 bit), and it seems to cut the lower bits on mips in this case.

When I change the return type in the definition ([`healpix_utils.c`](https://github.com/astropy/astropy-healpix/blob/528292bc2bff01563d6c696880020345cde88d40/cextern/astrometry.net/healpix-utils.c#L142-L143)) to `int`, this problem would be solved. However, the change to `int64_t` was probably done in fd48d256364a33d37950c41e95736db6810e9f33 on purpose.

IMO the declarations should be fixed here, and somehow a declaration should be included in `core_cython.pyx` -- just changing them to `int64_t` did _not_ solve the problem. And I am not really knowledgable in Cython to do that properly.

**Comment by lpsinger** [comment]
The latest release (0.3.1) now correctly declares the return type of `healpix_rangesearch_radec_simple` as `int64_t` in all places, but we still get an empty list from this function on MIPS (see https://buildd.debian.org/status/fetch.php?pkg=astropy-healpix&arch=mips&ver=0.3.1-1&stamp=1544321901&raw=0). @olebole, any ideas?

**Comment by lpsinger** [comment]
This was fixed in 0.4.
